packages/firestore/test/unit/specs/listen_spec.test.ts

```diff
@@ -737,4 +737,32 @@ describeSpec('Listens:', [], () => {
         .expectEvents(query, { added: [docB] });
     }
   );
+
+  specTest('Query is established in new primary tab', ['multi-client'], () => {
+    const query = Query.atPath(path('collection'));
+    const docA = doc('collection/a', 1000, { key: 'a' });
+    const docB = doc('collection/b', 2000, { key: 'b' });
+
+    // Client 1 and Client 3 listen to the same query. When client 1 shuts
+    // down, client 2 becomes primary and takes ownership of a query it
+    // did not previously listen to.
+    return client(0, /*withGcEnabled=*/ false)
+      .expectPrimaryState(true)
+      .userListens(query)
+      .watchAcksFull(query, 1000, docA)
+      .expectEvents(query, { added: [docA] })
+      .client(1)
```

Comment:
just to sanity-check, we need this client(1) line even though there are no steps?

Reply:
This starts up and initializes the second client. Comment added.

We probably don't need to do this here, but that would be a different test case: What happens when a client starts up, becomes primary, and there is already pending work. There's a TODO in SyncEngine to deal with this.